### PR TITLE
[dev-sci] Add SAS option for real-time parallel

### PR DESCRIPTION
**Pull request by BenjaminBlake-NOAA**
## DESCRIPTION OF CHANGES: 

- In this PR to the dev-sci branch of rrfs-workflow, the option for running with scale-aware SAS deep convection instead of GF deep convection is added.  It is unclear whether this option will be chosen for the final configuration of RRFS version 1 but it is currently being tested and evaluated in the real-time parallel.
- Two new CCPP suites are added: RRFS_sas, and RRFS_sas_nogwd (for the fire weather nest).  I am confused how the real-time parallel is running with the RRFS_sas CCPP suite without changes to ush/valid_param_vals.sh, scripts/exrrfs_make_ics.sh, and scripts/exrrfs_make_lbcs.sh, which have been updated to include RRFS_sas_nogwd for the fire weather nest.  I also had to add two new files to the parm directory for RRFS_sas_nogwd (the diag_table and field_table) so I would have thought new versions of these are also needed for RRFS_sas.
- The default values in FV3.input.yml_ensphy and input.nml.RRFSFW are now set to use SAS instead of GF.

## TESTS CONDUCTED: 

Tested in the real-time parallel.

### Machines/Platforms:

- WCOSS2
  - [x] Cactus/Dogwood
  - [ ] Acorn
- RDHPCS
  - [ ] Hera
  - [ ] Jet
  - [ ] Orion
  - [ ] Hercules

### Test cases: 

- [ ] Engineering tests
  - [ ] Non-DA engineering test
  - [ ] DA engineering test
    - [ ] Retro
    - [ ] Ensemble
    - [ ] Parallel
- [x] RRFS fire weather
- [x] RRFS_A:
- [ ] RRFS_B:
- [ ] RTMA:
- [ ] Others:

## CONTRIBUTORS (optional): 

@ShunLiu-NOAA @JiliDong-NOAA @MatthewPyle-NOAA

### CODE REVIEW

File: parm/FV3.input.yml_ensphy
Comment (JiliDong-NOAA) [inline parm/FV3.input.yml_ensphy:21]: gf_coldstart needs to be set to true here

File: parm/FV3.input.yml_ensphy
Comment (JiliDong-NOAA) [inline parm/FV3.input.yml_ensphy:140]: gf_coldstart needs to be set to true here

**Review by JiliDong-NOAA — COMMENTED** [review]
I only reviewed the change to REFS yaml file. gf_coldstart needs to set to true for mem1 and 4

**Comment by BenjaminBlake-NOAA** [comment]
Thanks @JiliDong-NOAA , I wasn't sure if we should change the default values in the workflow to run with SAS instead of GF but I will make those changes now.

**Comment by JiliDong-NOAA** [comment]
> Thanks @JiliDong-NOAA , I wasn't sure if we should change the default values in the workflow to run with SAS instead of GF but I will make those changes now.

@BenjaminBlake-NOAA The default value for gf_coldstart is false and you don't need to change anything else. We set the value to be true only when we have saSAS restart file to initialize a forecast with G-F convection, which are mem1 and 4. Hope this makes sense.

File: scripts/exrrfs_make_lbcs.sh
Comment (MatthewPyle-NOAA) [inline scripts/exrrfs_make_lbcs.sh:1]: Adding RRFS_sas for completeness here also makes sense to me.

**Review by MatthewPyle-NOAA — COMMENTED** [review]
Left a few comments - things for getting this one started.  

Also should we add an RRFS_sas field_table and diag_table?

File: ush/valid_param_vals.sh
Comment (BenjaminBlake-NOAA) [reply ush/valid_param_vals.sh:1]: @MatthewPyle-NOAA Got it.  I'll add RRFS_sas to the list.

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA Yes we can add a new diag table and field table for RRFS_sas.  What files should I use for that?  For the RRFS_sas_nogwd files I started from rrfsens_phy0002, but now I am thinking that is not correct.  I think RRFS_sas and RRFS_sas_nogwd can have similar files.

**Comment by MatthewPyle-NOAA** [comment]
I know little about diag and table files - I'll need to figure out what is being used for RRFS, and how different versions vary.

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA It looks like the field and diag table is primarily sensitive to whether smoke and dust are being used, if CLM is being used, and a bit on the microphysics being used.  But I hadn't seen your comment above (I was comparing in the workflow parm directory).

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA  Got it.  So maybe the FV3_HRRR_gf files can also be used for RRFS_sas, since it includes smoke/dust and everything else is the same besides the convection scheme.  We should confirm this with @ShunLiu-NOAA

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA @ShunLiu-NOAA I retrieved the field_table file from rrfs/v0.9.1/rrfs_na, which is being used for the real-time parallel, and that one is added to the parm directory as field_table.RRFS_sas and field_table.RRFS_sas_nogwd.  

The diag_table file being used for the deterministic forecast in the real-time parallel is similar to diag_table.FV3_HRRR_gf_clm (which I think is the one we had been using previously - I mistakenly thought we had been using FV3_HRRR_gf instead), so I copied that one to diag_table.RRFS_sas.  For diag_table.RRFS_sas_nogwd, I copied that from diag_table.FV3_HRRR_gf_nogwd.  

If what I did is not correct let me know - thanks.

**Review by ShunLiu-NOAA — APPROVED** [review]
Looks good to me.

**Review by JiliDong-NOAA — APPROVED** [review]

**Comment by JiliDong-NOAA** [comment]
Just a note that this PR has a dependency on the ufs-weather-model PR:

https://github.com/ufs-community/ufs-weather-model/pull/2405

**Comment by MatthewPyle-NOAA** [comment]
I'll say we put this one on hold until the model PR is merged, and we can update the model hash in the workflow

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA Jili's PR has been merged now.  Can you update the Externals.cfg file to point at the latest production/RRFSv1 hash for the model (#7813700)?  Then we can get this one merged.  Thanks!

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA Done.  Once this PR is merged I can open a PR to production/RRFS.v1 later today or tomorrow to get these changes added there too.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Looks good to me.  Thanks everyone for the work to get this problem resolved.

**Comment by BenjaminBlake-NOAA** [comment]
I just realized that more changes may be needed with this PR.  When generating the namelist with the workflow, parm/FV3.input.yml is used.  Since RRFS_sas is a new CCPP suite, I think we will need to add a section to parm/FV3.input.yml for RRFS_sas.  We didn't run into this issue with the real-time parallel because the namelist changes were made locally.  Sorry for not catching this sooner.

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA Can you fix it with an add-on PR?

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA Yes I can.  I am working on it now.

**Pull request merged**
